src/main/java/io/github/thebusybiscuit/slimefun4/api/items/ItemGroup.java

```diff
@@ -184,7 +185,7 @@ public void add(@Nonnull SlimefunItem item) {
             return;
         }
 
-        if (isRegistered() && !item.getAddon().getName().equals(this.addon.getName())) {
+        if (isRegistered() && !item.getAddon().getName().equals(this.addon.getName()) && !this.allowOtherAddonItems) {
```

Comment:
This should NOT use the field but rather the getter, otherwise overriding that method would have no effect.
And since you did not declare that method final, it can in fact be overridden.